nixos/doc/manual/release-notes/rl-2505.section.md

```diff
@@ -127,6 +127,10 @@
 
 - `networking.wireguard` now has an optional networkd backend. It is enabled by default when `networking.useNetworkd` is enabled, and it can be enabled alongside scripted networking with `networking.wireguard.useNetworkd`. Some `networking.wireguard` options have slightly different behavior with the networkd and script-based backends, documented in each option.
 
+- `buildDotnetModule` now uses JSON-based lockfiles instead of nix. Nix-based lockfiles were kept for compatibility however they should be migrated to JSON.
```

Comment:
Some ideas for alternative wording:

```suggestion
- `buildDotnetModule` now uses JSON-based instead of nix-based lockfiles.
  - Support for nix-based lockfiles was kept for compatibility, however this support will be dropped in the future.
```